Required changes for panda_moveit_config

rhaschke opened the pull request:
This merge commit gathers required changes for https://github.com/ros-planning/panda_moveit_config/issues/96. The individual PRs are listed in #193.

rhaschke committed:
Drop re-publishing joint_states_desired

rickstaa commented on franka_gazebo/config/franka_hw_sim.yaml:
```diff
@@ -1,9 +1,9 @@
+  arm_id: $(arg arm_id)
 
   finger1:
     gains: { p: 100, i: 25, d: 20 }
```
Just a quick reminder for ourselves that these PID gains need to be improved before this pull request is merged (see https://github.com/frankaemika/franka_ros/pull/173#issuecomment-938507534 for more information).

rickstaa replied:
Based on https://github.com/frankaemika/franka_ros/pull/173#issuecomment-964235123 this commit should be removed.

rhaschke committed:
Use relative includes

rhaschke committed:
Define hand_tcp frame

... in the center between and aligned with the fingers

rhaschke committed:
Rename panda_arm.xacro -> arm.xacro

... and xacro:panda_arm -> xacro:arm
to be consistent with hand.xacro

rhaschke committed:
Unify use of arm_id (arm.xacro) and ns (hand.xacro) to prefix

IMO, prefix much better describes the purpose of these two variables
in the present context, namely providing a name prefix to joints and links.

rhaschke committed:
Use both, detailed meshes and coarse collision geometries

They serve different purposes:
- coarse collision geometries mimic the internal self-collision checking
  of the real-robot controller
- mesh models should be use for collision checking with environmental objects

rhaschke committed:
Add detailed collision meshes for fingers

rhaschke committed:
Unify real-world and Gazebo URDF descriptions

- Declare inertial properties in a yaml file
- Use xacro:inertial macro

rhaschke committed:
Add dynamics parameters for joints from panda_gazebo

rhaschke committed:
Distinguish link8 and link8_sc

rhaschke committed:
Merge PRs #199 (modular-xacro), #191 (drop-joint-states-desired) and #194 (tool-frame) into develop

rickstaa commented:
@rhaschke, it appears that many of the changes we've suggested in this pull request are already in some form within the `develop` branch. Considering this, would it be advisable to close this draft?

rhaschke commented:
Closing, as most of the individual commits are merged upstream.
